packages/core/src/2d/sprite/Sprite.ts

```diff
@@ -0,0 +1,232 @@
+import { Vector2, Vector4, Rect } from "@oasis-engine/math";
+import { RefObject } from "../../asset/RefObject";
+import { Engine } from "../../Engine";
+import { Texture2D } from "../../texture";
+
+export class Sprite extends RefObject {
+  private static _tempVec2: Vector2 = new Vector2();
+
+  private static _VERTICES_FLAG = 0x1;
+  private static _UV_FLAG = 0x2;
+  private static _TRIANGLES_FLAG = 0x4;
+  /** Sprite._VERTICES_FLAG | Sprite._UV_FLAG | Sprite._TRIANGLES_FLAG */
+  private static _VER_UV_TRI_FLAG = 0x7;
+
+  /** The array containing sprite mesh triangles. */
+  public triangles: number[] = [];
+  /** The base texture coordinates of the sprite mesh. */
+  public uv: Vector2[] = [new Vector2(), new Vector2(), new Vector2(), new Vector2()];
+  /** The array containing sprite mesh vertex positions. */
+  public vertices: Vector2[] = [new Vector2(), new Vector2(), new Vector2(), new Vector2()];
+
+  /** The reference to the used texture. */
+  private _texture: Texture2D = null;
+  /** The rectangle this sprite uses on its texture. */
+  private _textureRect: Rect = new Rect();
+  /** Location of the sprite's center point in the rect on the original texture, specified in pixels. */
+  private _pivot: Vector2 = new Vector2();
+  /** Location of the sprite on the original texture, specified in pixels. */
+  private _rect: Rect = new Rect();
+  /** The number of pixels in the sprite that correspond to one unit in world space. */
+  private _pixelsPerUnit: number = 100;
+  /** The dirty flag to determine whether refresh buffer. */
+  private _dirtyFlag: number = Sprite._VER_UV_TRI_FLAG;
+
+  /**
+   * The reference to the used texture.
+   */
+  get texture(): Texture2D {
+    return this._texture;
+  }
+
+  set texture(value: Texture2D) {
+    if (this._texture !== value) {
+      this._texture = value;
+    }
+  }
+
+  /**
+   * The rectangle this sprite uses on its texture.
+   */
+  get textureRect(): Rect {
+    return this._textureRect;
+  }
+
+  set textureRect(value: Rect) {
+    if (this._textureRect !== value) {
+      this._textureRect.x = value.x;
+      this._textureRect.y = value.y;
+      this._textureRect.width = value.width;
+      this._textureRect.height = value.width;
+    }
+  }
+
+  /**
+   * Location of the sprite's center point in the rect on the original texture, specified in pixels.
+   */
+  get pivot(): Vector2 {
+    return this._pivot;
+  }
+
+  set pivot(value: Vector2) {
```

Comment:
If develop change the pivot and then call vertices still get old data until next farme.